lib/core/src/Cardano/Wallet/DB/Sqlite/CheckpointsOld.hs

```diff
@@ -225,19 +234,20 @@ mkStoreCheckpoints wid =
     Database operations
 -------------------------------------------------------------------------------}
 selectAllCheckpoints
-    :: forall s. PersistState s
+    :: forall s. PersistAddressBook s
     => W.WalletId
     -> SqlPersistT IO [(W.Slot, W.Wallet s)]
 selectAllCheckpoints wid = do
     cps <- fmap entityVal <$> selectList
         [ CheckpointWalletId ==. wid ]
         [ Desc CheckpointSlot ]
-    fmap catMaybes $ forM cps $ \cp -> do
+    -- FIXME LATER during ADP-1043: Presence of these tables?
```

Comment:
would be good to have check on if the tables exist and only after they are proceed :+1:

Reply:
Good point. I have added a commit where a new error type `ErrBadFormat` is introduced, which I intend to eventually cover all errors related to malformed database files. As a first step, it absorbs the `fromJust` case here.